File: MovieTicketBooking/Movie Booking Java/src/main/java/org/example/controllers/PaymentController.java
package org.example.controllers;

import lombok.AllArgsConstructor;
import org.example.model.Booking;
import org.example.service.BookingService;
import org.example.service.PaymentService;

@AllArgsConstructor
public class PaymentController {

    private final PaymentService paymentService;
    private final BookingService bookingService;

    public void onPaymentSuccess(String bookingId, String userId) {

        Booking booking = bookingService.getBooking(bookingId);
        bookingService.confirmBooking(booking, userId);
    }

    public void onPaymentFailure(String bookingId, String userId) {

        Booking booking = bookingService.getBooking(bookingId);
        paymentService.onPaymentFailed(booking, userId);
    }
}

File: MovieTicketBooking/Movie Booking C#/Controllers/PaymentController.cs
using MovieTicketBooking.Services;

namespace MovieTicketBooking.Controllers
{

    public class PaymentController
    {
        BookingService _bookingService;
        PaymentService _paymentService;
        public PaymentController(BookingService bookingService, PaymentService paymentService)
        {
            _bookingService = bookingService;
            _paymentService = paymentService;
        }
        public void ProcessPaymentSuccess(string user, string bookingId)
        {
            _bookingService.ConfrimBooking(bookingId, user);
        }
        public void ProcessPaymentFailed(string user, string bookingId)
        {
           _paymentService.ProcessPaymentFailed(user, bookingId);
        }
    }
}